library/core/src/iter/range.rs

```diff
@@ -484,6 +485,53 @@ impl Step for char {
     }
 }
 
+#[unstable(feature = "step_trait", reason = "recently redesigned", issue = "42168")]
+impl Step for AsciiChar {
+    #[inline]
+    fn steps_between(&start: &AsciiChar, &end: &AsciiChar) -> Option<usize> {
+        Step::steps_between(&start.as_u8(), &end.as_u8())
+    }
+
+    #[inline]
+    fn forward_checked(start: AsciiChar, count: usize) -> Option<AsciiChar> {
+        let end = Step::forward_checked(start.as_u8(), count)?;
+        if end <= 127 {
+            // SAFETY: ASCII characters are valid for range 0..=127
+            Some(unsafe { AsciiChar::from_u8_unchecked(end as u8) })
+        } else {
+            None
+        }
```

Comment:
Isn't this just `AsciiChar::from_u8(end)`?

Reply:
For some reason I genuinely believed that this method *didn't* exist when I made this PR, but I checked the history and it did. 😕

I'll use that.